### PR TITLE
[NeuralChat] enable gramma check and query polish to enhance RAG performance

### PR DESCRIPTION
**Pull request by XuhuiRen**
## Type of Change

enable polish for the user input query

help to revise the gramma error and polish the user query to make it easy to be understood.

Could help improve the retrieval accuracy based on the initial result from Intel Lab with only a simple word check.

same idea could be found at https://arxiv.org/abs/2305.14283

### CODE REVIEW

File: intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/query_explainer.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    def polish_query(self, model_name, query):
+        """Using the LLM to polish the user query to fix the gramma errors and clarify the user query."""
+        prompt = polish_query_prompt(query)
+        params = {}
+        params["model_name"] = model_name
+        params["prompt"] = prompt
+        params["temperature"] = 0.1
+        params["top_k"] = 3
+        params["max_new_tokens"] = 512
```
Comment (hshen14) [inline intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/query_explainer.py:36]: why limit to 512 for max length, and similar to temperature/top_k?

Comment (XuhuiRen) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/query_explainer.py:36]: this is only use for user query, which could not be too long. So, i set 512 for a considerable length tolerance. temperature value and top-k value is verified effective on the local machine. There is no general guideline for these settings. select the same way as intent detection to set these values.

Comment (hshen14) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/query_explainer.py:36]: the query should support up to the max LLM context length? what's the side effect, if the query is longer than 512?

Comment (XuhuiRen) [reply intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/detector/query_explainer.py:36]: in general, the user query should be less than 512 token length. we do not expect the user to change this parameter in RAG application. if there is some special use case, we can change this value

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**